File: 0443-string-compression/0443-string-compression.cpp
int init=[] {ios_base::sync_with_stdio(false);cin.tie(0);return 0;}();
class Solution {
public:
    int compress(std::vector<char>& chars) {
        int i = 0, res = 0, n = chars.size();
        
        while (i < n) {
            char letter = chars[i];
            int count = 0;

            while (i < n && chars[i] == letter) {
                count++;
                i++;
            }
            
            chars[res] = letter;
            res++;
            
            if (count > 1) {
                for (char digit : std::to_string(count)) {
                    chars[res] = digit;
                    res++;
                }
            }
        }
        return res;
    }
};

File: 0443-string-compression/0443-string-compression.py
class Solution:
  def compress(self, chars: List[str]) -> int:
    i = res = 0

    while i < len(chars):
      letter = chars[i]
      count = 0
      while i < len(chars) and chars[i] == letter:
        count += 1
        i += 1
      chars[res] = letter
      res += 1
      if count > 1:
        for digit in str(count):
          chars[res] = digit
          res += 1

    return res